web/ui/react-app/src/pages/flags/Flags.tsx

```diff
@@ -14,18 +17,108 @@ interface FlagsProps {
   data?: FlagMap;
 }
 
+const compareAlphaFn = (keys: boolean, reverse: boolean) => (
+  [k1, v1]: [string, string],
+  [k2, v2]: [string, string]
+): number => {
+  const a = keys ? k1 : v1;
+  const b = keys ? k2 : v2;
+  const reverser = reverse ? -1 : 1;
+  return reverser * a.localeCompare(b);
+};
+
+const boldSubstring = (s: string, subs: string) => {
+  if (subs === '' || s.indexOf(subs) < 0) {
+    return <span>{s}</span>;
+  }
+  return <span dangerouslySetInnerHTML={{ __html: s.split(subs).join(subs.bold()) }} />;
```

Comment:
We should be really careful whenever using `dangerouslySetInnerHTML`, as it opens us up to XSS.

In this case, if I start Prometheus like this:

    ./prometheus --web.page-title="foo<i>bar</i>"

...it will actually show up as italic.

We have one other place where we use `dangerouslySetInnerHTML` here: https://github.com/prometheus/prometheus/blob/d35cf369f213b893d88c52482978b6849891eb26/web/ui/react-app/src/pages/graph/ExpressionInput.tsx#L112

Could you use the same `sanitizeHTML()` call as there?

But going further, we do the searching there using a fuzzy algorithm: https://github.com/prometheus/prometheus/blob/d35cf369f213b893d88c52482978b6849891eb26/web/ui/react-app/src/pages/graph/ExpressionInput.tsx#L73-L78

For consistency, should we use the same approach here rather than doing our own search (and bold substring) processing?